File: templates/problems/13_dp_1d/03_house_robber/python/solution.py
def solve_rob(nums: list[int]) -> int:
    """Return the maximum amount that can be robbed from non-adjacent houses.

    Time: O(n), Space: O(1)
    """
    prev, curr = 0, 0
    for num in nums:
        prev, curr = curr, max(curr, prev + num)
    return curr

File: templates/problems/13_dp_1d/03_house_robber/typescript/solution.ts
/**
 * Return the maximum amount that can be robbed from non-adjacent houses.
 * Time: O(n), Space: O(1)
 */
export function solveRob(nums: number[]): number {
    let prev = 0;
    let curr = 0;
    for (const num of nums) {
        [prev, curr] = [curr, Math.max(curr, prev + num)];
    }
    return curr;
}
